src/containers/linking/LinkingSagas.js

```diff
@@ -0,0 +1,101 @@
+/*
+ * @flow
+ */
+
+import {
+  all,
+  delay,
+  call,
+  put,
+  select,
+  takeEvery,
+  takeLatest,
+} from '@redux-saga/core/effects';
+import { List, fromJS } from 'immutable';
+import {
+  SearchApiActions,
+  SearchApiSagas,
+} from 'lattice-sagas';
+import type { SequenceAction } from 'redux-reqseq';
+
+import Logger from '../../utils/Logger';
+import {
+  SEARCH_LINKED_ENTITY_SETS,
+  searchLinkedEntitySets,
+} from './LinkingActions';
+import { isNonEmptyString } from '../../utils/LangUtils';
+
+const LOG = new Logger('EntitySetSagas');
+
+const { searchEntitySetData } = SearchApiActions;
+const { searchEntitySetDataWorker } = SearchApiSagas;
+
+function* searchLinkedEntitySetsWorker(action :SequenceAction) :Generator<*, *, *> {
+
+  try {
+    yield put(searchLinkedEntitySets.request(action.id, action.value));
+
+    const { entitySetIds, searchFieldValues } = action.value;
+
+    const searchFields = searchFieldValues
+      .filter((searchField :Object) => isNonEmptyString(searchField.value))
+      .map((searchField :Object) => ({
+        property: searchField.propertyTypeId,
+        searchTerm: searchField.value,
+      }));
+
+    const searchOptions = {
+      searchFields,
+      start: 0,
+      maxHits: 1000,
+    };
+
+    const callMap = {};
+    entitySetIds.forEach((entitySetId :UUID) => {
+      callMap[entitySetId] = call(
+        searchEntitySetDataWorker,
+        searchEntitySetData({ entitySetId, searchOptions })
+      );
+    });
```

Comment:
Oh shit. I've been doing this with an array of calls instead of an object this whole time. Did not know this was available 🤦‍♂